### PR TITLE
Improvement/title: margin and padding

### PR DESCRIPTION
**Pull request by billyxs**

### CODE REVIEW

File: packages/title/src/Title.js [hunk 1]
```diff
@@ -2,7 +2,11 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { TextBase } from '@hixme-ui/text'
 
-export const TitleBase = TextBase.withComponent('h1')
+export const TitleBase = TextBase.withComponent('h1').extend`
+  margin: ${props => props.margin || '0px'};
+  padding: ${props => props.padding || '0px'};
+`
```
Comment (seesleestak) [inline packages/title/src/Title.js]: Cool, didn't know you could do this.

File: packages/title/src/Title.js [hunk 2]
```diff
@@ -2,7 +2,11 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { TextBase } from '@hixme-ui/text'
 
-export const TitleBase = TextBase.withComponent('h1')
+export const TitleBase = TextBase.withComponent('h1').extend`
+  margin: ${props => props.margin || '0px'};
```
Comment (seesleestak) [inline packages/title/src/Title.js]: Also, this syntax is much cleaner than what I've been doing. I like it.

File: packages/title/src/Title.js [hunk 3]
```diff
@@ -2,7 +2,11 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { TextBase } from '@hixme-ui/text'
 
-export const TitleBase = TextBase.withComponent('h1')
+export const TitleBase = TextBase.withComponent('h1').extend`
```
Comment (seesleestak) [inline packages/title/src/Title.js]: Cool, didn't know you could do this.

Comment (billyxs) [reply packages/title/src/Title.js]: yeah, I'm really impressed with how composable their components are.

**Pull request merged**